types/sinon/index.d.ts

```diff
@@ -1673,6 +1673,16 @@ declare namespace Sinon {
                     | (TType[K] extends (...args: any[]) => infer R ? R : TType[K]);
             },
         ): SinonStubbedInstance<TType>;
+
+        /**
+         * Defines a property on the given object which will be torn down when
+         * the sandbox is restored
+         */
+        define(
+            obj: object,
+            key: PropertyKey,
```

Comment:
> Purposely doesn't use generics since you should be able to define properties the object's type doesn't have

@43081j hey, there's some inconvenience that I believe is caused by the lack of generics. 

The property passed to `.define()` might or might not be declared in the class definition. In my case, the properties are declared, and I'd like the IDE to be able to jump to the property definition.

With `sandbox.replace()`, the IDE understands that the argument is an actual property: 

<img width="760" alt="image" src="https://github.com/DefinitelyTyped/DefinitelyTyped/assets/5655837/938e4c8e-15db-4c9b-9c82-16360f571f52">

And with `sandbox.define()`, it doesn't. It's just a string.

---

I'm not well-versed in how these type definitions work, please help me understand. 
Would it make sense to declare this type as a union of `PropertyKey` and  `TKey extends keyof T`?

Reply:
I verified that property discovery in WebStorm works well when the type is declared as `keyof T | PropertyKey`.

Here's PR with a corresponding change: 
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67433